File: Sample Algorithms/Exam Study/Recursion Exercies/7.c
// Write a recursive function that arranges the elements in the array in order

#include <stdio.h>
#define SWAPPED 1

void Swap(int* a,int* b){
  int temp = *a;
  *a = *b;
  *b = temp;
}

void Sort(int* array,int size,int current,int flag){
    if (current < 1){
        if(flag){
            return Sort(array,size,size-1,0);
        }
        return;
    }
    if(array[current] < array[current-1]){
        Swap(&array[current],&array[current-1]);
        return Sort(array,size,current,SWAPPED);
    }
    return Sort(array,size,--current,flag);
}



void main() {
    int arr[] = {1,23,6456,-22,4324,121};
    int arraySize = (sizeof(arr)/sizeof(arr[0]));
    Sort(arr,arraySize,arraySize-1,0);
    for(int i = 0; i < arraySize;i++){
      printf("[%d]",arr[i]);
    }
}

File: Sample Algorithms/Exam Study/Recursion Exercies/7.py
# Write a recursive function that arranges the elements in the array in order

def Sort(arr,n,flag):
    if(n < 1):
        if(flag == 1):
            return Sort(arr,len(arr)-1,0)
        return
    if(arr[n] < arr[n-1]):
        arr[n],arr[n-1] = arr[n-1],arr[n]
        return Sort(arr,n,1)
    return Sort(arr,n-1,flag)
        

a = [1,-212321,12412421,112,344,-55,11]
Sort(a,len(a)-1,0)
print("Sorted array:",a)